source/mainnet/smart-contracts/onboarding-guide-solana-developers/overview.rst

```diff
@@ -0,0 +1,112 @@
+.. _solana-overview:
+
+========
+Overview
+========
+
+This overview outlines the main differences of account models and smart contract functionality between Solana and Concordium.
+
+Accounts
+========
+
+Solana accounts are somewhat similar to files and can hold arbitrary data.
+Some accounts are marked as `executable` and contain program code.
+These accounts themselves are immutable, but programs stored in their data can modify data in accounts they own.
+
+Concordium, on the other hand, has a strict separation between user accounts and smart contracts.
+Both accounts and smart contracts have CCD balances associated with them.
+Moreover, smart contracts can have state -- structured data associated with the contract.
+Unlike Solana, this data can be updated directly only by the smart contract code it belongs to.
+The only way of interacting with the contract state from outside is through the interface that the smart contract provides.
+This interface consists of entrypoints.
+
+Another important difference is that transferring CCD to a non-existing account or interacting with a non-existing smart contract address on Concordium will fail.
+
+.. _solana-programs:
+
+Smart contracts and programs
+============================
+
+Concordium `smart contracts` serve the same purpose as `programs` on Solana.
+However, some details are different in how these two are structured.
+
+Solana programs take a list of accounts as input.
+Some of these accounts can be used to store persistent data, that is, data that is stored after the contract invocation is completed.
+Concordium uses smart contract states for that purpose.
+Each contract can write to its own state directly.
+Modifying the state of other smart contracts is only possible by calling their entrypoints.
```

Comment:
Is that different from Solana?

Reply:
I guess, it's kind of the same, only Solana programs can write to accounts they own, and interaction with the program happens through entrypoints.
My point is that Solana accounts, that are used for storing arbitrary state don't have an associated interface to access them, it could be any program that owns them and the owner can be changed. That is, any program can read data from any account.

Another point is that SOL (the naitive currency) can be transferred to any account, apparently also to a program account as well (can't find whether these SOLs can be recovered).